dot/sync/chain_processor_test.go

```diff
@@ -0,0 +1,607 @@
+				babeVerifier: mockBabeVerifier,
+			},
+			err: errors.New("could not verify block: nil header"),
+		},
+		{
+			name: "base case",
+			fields: fields{
+				babeVerifier: mockBabeVerifier,
+			},
+			args: args{header: &types.Header{
+				Number: 0,
+			}},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			s := &chainProcessor{
+				babeVerifier: tt.fields.babeVerifier,
+			}
+			err := s.handleHeader(tt.args.header)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_chainProcessor_handleJustification(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+
+	mockFinalityGadget := NewMockFinalityGadget(ctrl)
+	mockFinalityGadget.EXPECT().VerifyBlockJustification(gomock.AssignableToTypeOf(common.Hash{}),
+		gomock.AssignableToTypeOf([]byte{})).DoAndReturn(func(_ common.Hash, justification []byte) error {
+		if len(justification) < 2 {
+			return errors.New("error")
+		}
+		return nil
+	}).Times(3)
+	mockBlockState := NewMockBlockState(ctrl)
+	mockBlockState.EXPECT().SetJustification(gomock.AssignableToTypeOf(common.Hash{}),
+		gomock.AssignableToTypeOf([]byte{})).DoAndReturn(func(_ common.Hash, justification []byte) error {
+		if bytes.Equal(justification, []byte(`xx`)) {
+			return errors.New("fake error")
+		}
+		return nil
+	}).Times(2)
+
+	type fields struct {
+		blockState     BlockState
+		finalityGadget FinalityGadget
+	}
+	type args struct {
+		header        *types.Header
+		justification []byte
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+	}{
+		{
+			name: "nil justification and header",
+		},
+		{
+			name: "invalid justification",
+			fields: fields{
+				finalityGadget: mockFinalityGadget,
+			},
+			args: args{
+				header: &types.Header{
+					Number: 0,
+				},
+				justification: []byte(`x`),
+			},
+		},
+		{
+			name: "set justification error",
+			fields: fields{
+				blockState:     mockBlockState,
+				finalityGadget: mockFinalityGadget,
+			},
+			args: args{
+				header: &types.Header{
+					Number: 0,
+				},
+				justification: []byte(`xx`),
+			},
+		},
+		{
+			name: "base case set",
+			fields: fields{
+				blockState:     mockBlockState,
+				finalityGadget: mockFinalityGadget,
+			},
+			args: args{
+				header: &types.Header{
+					Number: 0,
+				},
+				justification: []byte(`1234`),
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			s := &chainProcessor{
+				blockState:     tt.fields.blockState,
+				finalityGadget: tt.fields.finalityGadget,
+			}
+			s.handleJustification(tt.args.header, tt.args.justification)
+		})
+	}
+}
+
+func Test_chainProcessor_processBlockData_nilBlockData(t *testing.T) {
+	t.Parallel()
+
+	s := &chainProcessor{}
+	err := s.processBlockData(nil)
+	assert.EqualError(t, err, errors.New("got nil BlockData").Error())
```

Comment:
```suggestion
	assert.EqualError(t, err, "got nil BlockData")
```

Also is this a sentinel error? Maybe we could assert it with `assert.ErrorIs` instead

Reply:
Good point, updated to check sentinel error.